Sources/SwifterSwift/UIKit/UIViewExtensions.swift

```diff
@@ -579,7 +578,72 @@ public extension UIView {
     func ancestorView<T: UIView>(withClass name: T.Type) -> T? {
         return ancestorView(where: { $0 is T }) as? T
     }
+}
+
+// MARK: - findConstraint and friends
+public extension UIView {
+    /// SwifterSwift: Return constraints for this view. This must enumerate superviews.
+    var allConstraints: [NSLayoutConstraint] {
+        // https://stackoverflow.com/questions/47053727
+
+        // build array of self and all superviews
+        var views = [self]
+        var current = self
+        while let superview = current.superview {
+            views.append(superview)
+            current = superview
+        }
+
+        // now collect constraints affecting this view
+        return views.flatMap { $0.constraints }.filter {
+            $0.firstItem as? UIView == self || $0.secondItem as? UIView == self
+        }
+    }
+
+    /// SwifterSwift: Search constraints for this view until we find one for the given attribute.
+    ///
+    /// - Parameter attribute: the attribute to find
+    /// - Returns: matching constraint
+    func findConstraint(attribute: NSLayoutConstraint.Attribute) -> NSLayoutConstraint? {
+        // This is a bit subtle. While looking for .height, it's entirely
+        // possible that firstAttribute is .height but firstItem is not self
+        // (secondItem would be self in that case). That's not what we want to
+        // return.
+        allConstraints.first {
```

Comment:
Maybe it does make sense to not reuse `allConstraints` here, because if we implement the search here we can optimize and return early if we find then right away. What do you think?

Reply:
The constraint properties are the useful extensions here. allConstraints and findConstraint are just dependencies. The properties are unlikely to be performant because of how autolayout works, but we can certainly make them a bit faster by removing allConstraints entirely. I'll take a look.

Reply:
how about this?